service/controller/appcatalog/resource/appcatalogentry/create.go

```diff
@@ -140,6 +140,12 @@ func (r *Resource) newAppCatalogEntries(ctx context.Context, cr v1alpha1.AppCata
 		for _, entry := range entries {
 			name := fmt.Sprintf("%s-%s-%s", cr.Name, entry.Name, entry.Version)
 
+			restrictions, err := r.getRestrictions(ctx, key.AppCatalogStorageURL(cr), entry.Name, entry.Version)
```

Comment:
I think this should be `getMetadata()` and then parse the restrictions data from it separately.  WDYT?